lib/ReactViews/Map/Panels/SharePanel/SharePanel.tsx

```diff
@@ -0,0 +1,667 @@
+      window.addEventListener("afterprint", this.afterBrowserPrint, false);
+
+      const handlePrintMediaChange = (evt: MediaQueryListEvent) => {
+        if (evt.matches) {
+          this.beforeBrowserPrint();
+        } else {
+          this.afterBrowserPrint();
+        }
+      };
+
+      if (window.matchMedia) {
+        const matcher = window.matchMedia("print");
+        matcher.addEventListener("change", handlePrintMediaChange); // MediaQueryList.addListener() is deprecated replaced with addEventListener()
+        this._unsubscribeFromPrintMediaChange = function() {
+          matcher.removeEventListener("change", handlePrintMediaChange);
+        };
+      }
+
+      this._oldPrint = window.print;
+      window.print = () => {
+        this.print();
+      };
+    }
+
+    // TODO: Remove after review:
+    // THIS MOVED TO onIncludeStoryInShareClicked() and is now implemented with a setState callback
+    // Listen to the includeStoryInShare property of viewState, generate the share URL again if this changes
+    // This allows the share URL to be updated when users change the 'Include Story in Share?'checkbox in the SharePanel component.
+    // this.updateShareUrlWhenStoryOptionChanged = reaction(
+    //   () => this.state.includeStoryInShare,
+    //   () => {
+    //     this.updateForShortening();
+    //   }
+    // );
+  }
+  print() {
+    throw new Error("Method not implemented.");
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("beforeprint", this.beforeBrowserPrint, false);
+    window.removeEventListener("afterprint", this.afterBrowserPrint, false);
+    if (this._unsubscribeFromPrintMediaChange) {
+      this._unsubscribeFromPrintMediaChange();
+    }
+
+    if (this._oldPrint) {
+      window.print = this._oldPrint;
+    }
+
+    // TODO: Remove after review:
+    // Cleanup reaction
+    // this.updateShareUrlWhenStoryOptionChanged &&
+    //   this.updateShareUrlWhenStoryOptionChanged();
+  }
+
+  beforeBrowserPrint() {
+    const { t } = this.props;
+    this.afterBrowserPrint();
+    this._message = document.createElement("div");
+    this._message.innerText = t("share.browserPrint", {
+      appName: this.props.terria.appName
+    });
+    window.document.body.insertBefore(
+      this._message,
+      window.document.body.childNodes[0]
+    );
+  }
+
+  afterBrowserPrint() {
+    if (this._message) {
+      window.document.body.removeChild(this._message);
+      this._message = undefined;
+    }
+    this.changeOpenState(true);
+  }
+
+  advancedIsOpen() {
+    return this.state.advancedIsOpen;
+  }
+
+  toggleAdvancedOptions = (e: React.MouseEvent<HTMLButtonElement>) => {
+    this.setState((prevState: SharePanelState) => ({
+      advancedIsOpen: !prevState.advancedIsOpen
+    }));
+  };
+
+  updateForShortening() {
+    // TODO: Make sure that this.setState is accessible, may need to this.updateForShortening = this.updateForShortening.bind(this) or change to arrow function
```

Comment:
Should be accessible - it looks like this is only called in methods which are bound to `this` in the constructor

Reply:
Cheers, forgot to remove this comment after binding to this